### PR TITLE
[NodeTypeResolver] Move namespaced name check into specific NameTypeResolver

### PR DESCRIPTION
**Pull request by samsonasik**
Ref review comment https://github.com/rectorphp/rector-src/pull/6089#issuecomment-2199362950

`Name` can be a type, and we have `NameTypeResolver`

https://github.com/rectorphp/rector-src/blob/ef333502fb722ef471049b492a257781d02d0d62/src/NodeTypeResolver/NodeTypeResolver/NameTypeResolver.php#L25-L33

The `namespaced_name` attribute was only working on `NullableType` type, thats why PR https://github.com/rectorphp/rector-src/pull/5947 added to work recursively to get short name support, eg: alias name, which never be FullyQualified.

The `getType()` method works on recursive way in nodes:

https://github.com/rectorphp/rector-src/blob/ef333502fb722ef471049b492a257781d02d0d62/src/NodeTypeResolver/NodeTypeResolver.php#L126-L129

For `Expr` only, there is `getNativeType()` for `Expr` only, but for `getType()`, there is list supported type resolvers:

https://github.com/rectorphp/rector-src/tree/main/src/NodeTypeResolver/NodeTypeResolver

### CODE REVIEW

**Comment by TomasVotruba** [comment]
I see. Could you add failing fixture to verify this is needed?

**Comment by samsonasik** [comment]
I will try.

**Comment by samsonasik** [comment]
I added test for existing class short name node short name https://github.com/rectorphp/rector-src/pull/6093/commits/ac85ebfb59334d2d1b741956ce3d27a7c3a64fc3 

without this, it will got non existing object type, which actually exists

```diff
+Rector\StaticTypeMapper\ValueObject\Type\NonExistingObjectType Object #5255 (
+    'className' => 'AnotherClass',
```

as pointed to `Rector\Tests\NodeTypeResolver\Source\AnotherClass`

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I think it is ready.

File: rules/DeadCode/Rector/If_/RemoveDeadInstanceOfRector.php [hunk 1]
```diff
@@ -158,10 +157,6 @@ private function isInstanceofTheSameType(Instanceof_ $instanceof): ?bool
             return null;
         }
 
-        if (! $instanceof->class instanceof FullyQualified) {
-            return null;
-        }
```
Comment (TomasVotruba) [inline rules/DeadCode/Rector/If_/RemoveDeadInstanceOfRector.php:163]: This must stay here, as any other expression should be skipped.

Comment (samsonasik) [reply rules/DeadCode/Rector/If_/RemoveDeadInstanceOfRector.php:163]: It already checked above

Comment (TomasVotruba) [reply rules/DeadCode/Rector/If_/RemoveDeadInstanceOfRector.php:163]: I see. Please send in separate PR

Comment (samsonasik) [reply rules/DeadCode/Rector/If_/RemoveDeadInstanceOfRector.php:163]: sure https://github.com/rectorphp/rector-src/pull/6096

**Comment by TomasVotruba** [comment]
I've just release new Rector 1.2.0 with this feature. 
Closing for now to keep focus, let's wait for real-life feedback and fixtures :)

**Pull request closed without merging**